issuance/issuance.go

```diff
@@ -650,12 +658,17 @@ func (i *Issuer) Prepare(req *IssuanceRequest) ([]byte, *issuanceToken, error) {
 
 	if req.IncludeCTPoison {
 		template.ExtraExtensions = append(template.ExtraExtensions, ctPoisonExt)
-	} else if req.SCTList != nil {
-		sctListExt, err := generateSCTListExt(req.SCTList)
+	} else if len(req.sctList) > 0 {
```

Comment:
A slight tweak that makes this a little easier to read. Feel free to take or leave this:

```go
	if req.IncludeCTPoison {
		template.ExtraExtensions = append(template.ExtraExtensions, ctPoisonExt)
	} else {
		if len(req.sctList) > 0 {
			if len(req.precertDER) == 0 {
				return nil, nil, errors.New("inconsistent request contains sctList but no precertDER")
			}

			sctListExt, err := generateSCTListExt(req.sctList)
			if err != nil {
				return nil, nil, err
			}

			template.ExtraExtensions = append(template.ExtraExtensions, sctListExt)
		} else {
			return nil, nil, errors.New("invalid request contains neither sctList nor precertDER")
		}
	}
```

Reply:
Thanks! I'm going to leave it; I think it's better to have the three cases as part of an `if` / `else if` / `else` group because I think they belong logically together:

 - precert
 - final cert
 - oops, neither one